pkg/discovery/dtofactory/general_builder.go

```diff
@@ -116,90 +122,103 @@ func (builder generalBuilder) getResourceCommoditiesSold(entityType metrics.Disc
-	if err != nil {
-		glog.Errorf("Failed to get %s used for %s %s: %s", resourceType, entityType, entityID, err)
-		return nil, fmt.Errorf("Failed to get %s used for %s %s: %s", resourceType, entityType, entityID, err)
-	}
-	usedValue := usedMetric.GetValue().(float64)
-	if converter != nil && converter.Convertible(resourceType) {
-		oldValue := usedValue
-		usedValue = converter.Convert(resourceType, usedValue)
-		glog.V(4).Infof("Convert %s used value from %f to %f for %s - %s", resourceType, oldValue, usedValue, entityType, entityID)
+	// check for the unit converter for cpu resources
+	if metrics.IsCPUType(resourceType) && converter == nil {
+		return nil, fmt.Errorf("missing cpu converter")
 	}
-	commBoughtBuilder.Used(usedValue)
 
-	capacityUID := metrics.GenerateEntityResourceMetricUID(entityType, entityID, resourceType, metrics.Capacity)
-	capacityMetric, err := builder.metricsSink.GetMetric(capacityUID)
+	commSoldBuilder := sdkbuilder.NewCommodityDTOBuilder(cType)
+
+	// set used value
+	usedValue, err := builder.metricValue(entityType, entityID,
+		resourceType, metrics.Used, converter)
 	if err != nil {
-		glog.Errorf("Failed to get %s capacity for %s %s: %s", resourceType, entityType, entityID, err)
-		return nil, fmt.Errorf("Failed to get %s capacity for %s %s: %s", resourceType, entityType, entityID, err)
+		return nil, fmt.Errorf("%s", err)
 	}
-	capacityValue := capacityMetric.GetValue().(float64)
-	if converter != nil && converter.Convertible(resourceType) {
-		oldValue := capacityValue
-		capacityValue = converter.Convert(resourceType, capacityValue)
-		glog.V(4).Infof("Convert %s capacity value from %f to %f for %s - %s", resourceType, oldValue, capacityValue, entityType, entityID)
 
+	commSoldBuilder.Used(usedValue)
+
+	// set capacity value
+	capacityValue, err := builder.metricValue(entityType, entityID,
+		resourceType, metrics.Capacity, converter)
+	if err != nil {
+		return nil, fmt.Errorf("%s", err)
 	}
-	commBoughtBuilder.Capacity(capacityValue)
+	commSoldBuilder.Capacity(capacityValue)
 
 	// set additional attribute
 	if commodityAttrSetter != nil && commodityAttrSetter.Settable(resourceType) {
-		commodityAttrSetter.Set(resourceType, commBoughtBuilder)
+		commodityAttrSetter.Set(resourceType, commSoldBuilder)
 	}
 	if commKey != "" {
-		commBoughtBuilder.Key(commKey)
+		commSoldBuilder.Key(commKey)
 	}
-	resourceCommoditySold, err = commBoughtBuilder.Create()
+	resourceCommoditySold, err = commSoldBuilder.Create()
 	if err != nil {
-		glog.Errorf("Failed to build commodity sold: %s", err)
 		return nil, err
 	}
 	return resourceCommoditySold, nil
 }
 
+func (builder generalBuilder) metricValue(entityType metrics.DiscoveredEntityType, entityID string,
+	resourceType metrics.ResourceType, metricProp metrics.MetricProp,
+	converter *converter) (float64, error) {
+	metricUID := metrics.GenerateEntityResourceMetricUID(entityType, entityID, resourceType, metricProp)
+	metric, err := builder.metricsSink.GetMetric(metricUID)
```

Comment:
log the error which is important for debugging

Reply:
Errors from these helper methods are logged in the main getResourceCommoditiesSold(...) and getResourceCommoditiesBought(...) methods using the entity Id as reference.